openedx/core/djangoapps/course_roles/permissions.py

```diff
@@ -0,0 +1,139 @@
+"""
+Permissions for course roles app.
+"""
+from enum import Enum, unique
+
+from django.utils.translation import gettext as _
+
+
+class PermissionData:
+    """
+    Data class for a permission.
+    """
+
+    def __init__(self, name, readable_name, description):
+        self.name = name
+        self.readable_name = readable_name
+        self.description = description
+
+
+@unique
+class CourseRolesPermission(Enum):
+    """
+    Enum of all user permissions, the values are the permissions names
+    in the course_roles_permission table in database.
```

Comment:
Before this gets merged, we really need some instructions here on what people are supposed to do when they want to add a permission. A long comment explaining it will do, but the lifecycle aspects of this are going to be easy for developers to mess up without clear guidance. I can just imagine someone deleting and re-building the permissions list with new IDs when they add a permission, causing something that looks like it works fine in tests, but ends up deleting everyone's existing permissions on production.